File: src/main/org/nlogo/prim/_fd1.java
package org.nlogo.prim;

import org.nlogo.agent.Turtle;
import org.nlogo.nvm.Command;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Syntax;

public final strictfp class _fd1
    extends Command {
  @Override
  public Syntax syntax() {
    return Syntax.commandSyntax("-T--", true);
  }

  @Override
  public void perform(Context context) {
    perform_1(context);
  }

  public void perform_1(Context context) {
    try {
      ((Turtle) context.agent).jump(1);
    } catch (org.nlogo.api.AgentException e) {
    } // NOPMD
    context.ip = next;
  }
}

File: src/main/prim/_fd1.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim

import org.nlogo.{ api, agent }
import org.nlogo.nvm.{ Command, Context }

class _fd1 extends Command {
  override def syntax =
    api.Syntax.commandSyntax("-T--", true)
  override def perform(context: Context) {
    perform_1(context)
  }
  def perform_1(context: Context) {
    try context.agent.asInstanceOf[agent.Turtle].jump(1)
    catch { case e: api.AgentException => } // ignore
    context.ip = next
  }
}
